### PR TITLE
feat: add phone number field to account settings

### LINKED ISSUE #701: [Email/SMS] Collect phone number on FE w/ validation
What feature is this part of?
"New Listing" Email/SMS Notifications #695 

Some questions we need to answer:
- [ ] Do we care about non-US numbers?
- [ ] How strict do we want to be with validation?
- [ ] Do we need editing phone number functionality?
- [ ] Do we need deleting phone number functionality?

### PR DESCRIPTION
**Pull request by plunkettgoogle**
## Issue

- Closes #701

## Description

Adds a PhoneField to the account/edit page.

Desktop:
localhost_3000_account_edit (3)

Mobile:
localhost_3000_account_edit (4)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/addresses technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Log in as test@example.com, navigate to account settings. Being able to actually update settings is blocked on https://github.com/CityOfDetroit/bloom/issues/744 currently.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes
- [ ] I have run `yarn generate:client` if I made backend changes
- [ ] I have exported any new pieces in ui-components

### CODE REVIEW

File: sites/public/pages/account/edit.tsx [hunk 1]
```diff
@@ -90,6 +93,24 @@ const Edit = () => {
     }
   }
 
+  const onPhoneSubmit = async (data: { phoneNumber: string }) => {
+    const { phoneNumber } = data
+    console.log("onPhoneSubmit(" + usToIntlPhone(phoneNumber) + ")")
```
Comment (GabeBenjamin) [inline sites/public/pages/account/edit.tsx:98]: nit: do we need this console.log?

**Comment by GabeBenjamin** [comment]
This isn't directly related to your PR, but it seems less than optimal to have each field be its own form. I wonder how often a user might click the wrong update button and not fully save whatever they change

**Review by GabeBenjamin — APPROVED** [review]

File: sites/public/pages/account/edit.tsx (hunk 1 above)
Comment (plunkettgoogle) [reply sites/public/pages/account/edit.tsx]: Nope, left over from debugging.

**Comment by plunkettgoogle** [comment]
> This isn't directly related to your PR, but it seems less than optimal to have each field be its own form. I wonder how often a user might click the wrong update button and not fully save whatever they change

Yeah, I've had this thought too. I'll open a FR.

**Pull request merged**